src-v5/carousel.tsx

```diff
@@ -49,12 +49,20 @@ export const Carousel = (props: CarouselProps): React.ReactElement => {
     );
     to && props.beforeSlide(slide, endSlide);
     !props.disableAnimation && setAnimation(true);
-
     setCurrentSlide(to ?? currentSlide);
     setTimeout(
       () => {
         if (!isMounted.current) return;
-        to && props.afterSlide(currentSlide);
+
+        let nextIndex = to ?? currentSlide;
+        if (nextIndex < 0) {
+          nextIndex += count;
+        }
+        if (nextIndex === count) {
+          nextIndex = 0;
+        }
```

Comment:
if the next index is the total number of slides that means we went backwards then forwards (while looping) which means we are at the first slide.